### PR TITLE
Fixed #26713 -- Documented resolving deprecation warnings before upgrade.

### PR DESCRIPTION
**Pull request by tomchristie**
Trac: https://code.djangoproject.com/ticket/26713

### CODE REVIEW

**Comment by tomchristie** [comment]
Related: https://github.com/django/django/pull/6733

File: docs/howto/upgrade-version.txt [hunk 1]
```diff
@@ -69,9 +94,9 @@ Testing
 =======
 
 When the new environment is set up, :doc:`run the full test suite
-</topics/testing/overview>` for your application. In Python 2.7+, deprecation
-warnings are silenced by default. It is useful to turn the warnings on so they
-are shown in the test output (you can also use the flag if you test your app
+</topics/testing/overview>` for your application. Again, it is useful to turn
```
Comment (timgraham) [inline docs/howto/upgrade-version.txt]: it's
turn on deprecation warnings
(also flow the "(you can.." bits to the previous line to keep consistent line breaks)

File: docs/howto/upgrade-version.txt [hunk 2]
```diff
@@ -42,6 +42,31 @@ released or if some of your dependencies are not well-maintained, some of your
...
 
+Resolving deprecation warnings
+==============================
+
+Before starting an upgrade it's important to resolve any existing deprecation
+warnings that are raised by your project. Working through these warnings prior
+to upgrading ensures that you're fully informed about areas of the code that
+need altering.
+
+In Python 2.7+, deprecation warnings are silenced by default. It is important
```
Comment (timgraham) [inline docs/howto/upgrade-version.txt]: I think we can drop "2.7+" at this point since 2.6 is mostly gone from the Django ecosystem.
It is -> It's

**Comment by timgraham** [comment]
It would be useful to mention that if you're using third-party apps, they might use deprecated APIs so that they can support multiple versions of Django, so you shouldn't worry about that or file bugs against those apps.

We could add a more complete guide of how third-party apps should handle multiple Django version support (essentially duplicating the information from "Deprecating warnings are no longer loud by default" in the 1.11 release notes), but that could be a separate ticket I suppose.

**Comment by tomchristie** [comment]
Thanks for the review @timgraham. Comments addressed.

> We could add a more complete guide of how third-party apps should handle multiple Django version support ... but that could be a separate ticket I suppose.

Agree - that's a somewhat different set of concerns.

**Comment by timgraham** [comment]
merged in 316395d82874beea6af08a873e91b1880d0b982a, thanks!

**Pull request closed without merging**